cloud/drift/status.go

```diff
@@ -16,24 +16,44 @@ import (
 const (
 	// ErrInvalidStatus represents an invalid drift status error.
 	ErrInvalidStatus errors.Kind = "invalid drift status"
+	// ErrInvalidFilterStatus represents an invalid drift status filter error.
+	ErrInvalidFilterStatus errors.Kind = "invalid filter stack status"
 )
 
-// Status of a drift.
-type Status uint8
+type (
+	// Status of a drift.
+	Status uint8
+
+	// FilterStatus represents a filter for drift statuses.
+	FilterStatus Status
+)
+
+const (
+	OK           Status = 1 << iota // OK status is used when the stack is not drifted.
+	Unknown                         // Unknown indicates the drift detection was not executed yet.
+	Drifted                         // Drifted status indicates the stack is drifted.
+	Failed                          // Failed status indicates the drift detection of the stack failed.
+	Unrecognized                    // Unrecognized indicates any drift status returned from TMC but not recognized by this client version.
+	lastStatus   = Failed
+)
 
 const (
-	OK         Status = 1 << iota // OK status is used when the stack is not drifted.
-	Unknown                       // Unknown indicates the drift detection was not executed yet.
-	Drifted                       // Drifted status indicates the stack is drifted.
-	Failed                        // Failed status indicates the drift detection of the stack failed.
-	lastStatus = Failed
+	// UnhealthyFilter status is used for filtering not OK drift status.
+	UnhealthyFilter FilterStatus = FilterStatus(^OK)
+	// HealthyFilter status is used for filtering for healthy statuses. Just [OK] for now.
+	HealthyFilter FilterStatus = FilterStatus(OK)
+	// NoFilter disables the filtering for statuses.
+	NoFilter FilterStatus = 0
 )
 
 // NewStatus creates a new stack drift status from a string.
-func NewStatus(str string) (Status, error) {
+func NewStatus(str string) Status {
 	var s Status
 	err := s.UnmarshalJSON([]byte(strconv.Quote(str)))
-	return s, err
+	if err != nil {
+		panic(errors.E(errors.ErrInternal, "Terramate must not fail when parsing drift statuses"))
```

Comment:
There are 3 implementations for this in deployment, stack and drift, and only two of them panic, the one for stack just continues. Maybe worth aligning them?

Reply:
it should never fail unless JSON encoding has issues encoding a quoted string.

Reply:
I will remove this panic in this and other cases